lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -573,51 +573,30 @@ Sample DistributionImplementation::getSample(const UnsignedInteger size) const
 /* Get one realization of the distribution */
 Point DistributionImplementation::getRealizationByInversion() const
 {
-  // Use CDF inversion in the 1D case
-  if (dimension_ == 1) return computeQuantile(RandomGenerator::Generate());
-  // Use conditional CDF inversion in the 1D case
-  Point point(0);
-  for (UnsignedInteger i = 0; i < dimension_; ++ i)
-  {
-    const Scalar u = RandomGenerator::Generate();
-    LOGINFO(OSS(false) << "i=" << i << ", u=" << u);
-    point.add(computeConditionalQuantile(u, point));
-    LOGINFO(OSS(false) << "i=" << i << ", u=" << u << ", point=" << point);
-  }
-  return point;
+  return getSampleByInversion(1)[0];
 }
 
 /* Get a numerical sample whose elements follow the distributionImplementation */
 Sample DistributionImplementation::getSampleByInversion(const UnsignedInteger size) const
 {
   // Use CDF inversion in the 1D case
   if (dimension_ == 1) return computeQuantile(RandomGenerator::Generate(size));
-  // Use conditional CDF inversion in the 1D case
-  Sample result(size, 0);
-  for (UnsignedInteger i = 0; i < dimension_; ++ i)
-  {
-    const Point u(RandomGenerator::Generate(size));
-    SampleImplementation q(size, 1);
-    q.setData(computeConditionalQuantile(u, result));
-    result.stack(q);
-  }
+  // Use conditional CDF inversion in the nD case
+  Sample result(size, dimension_);
+  for (UnsignedInteger i = 0; i < size; ++ i)
+    result[i] = computeSequentialConditionalQuantile(RandomGenerator::Generate(size));
   return result;
 }
 
 Sample DistributionImplementation::getSampleByQMC(const UnsignedInteger size) const
 {
-  const SobolSequence sequence(1);
+  static SobolSequence sequence(dimension_);
   // Use CDF inversion in the 1D case
-  if (dimension_ == 1) return computeQuantile(sequence.generate(size).getImplementation()->getData());
-  // Use conditional CDF inversion in the 1D case
-  Sample result(size, 0);
-  for (UnsignedInteger i = 0; i < dimension_; ++ i)
-  {
-    const Point u(sequence.generate(size).getImplementation()->getData());
-    SampleImplementation q(size, 1);
-    q.setData(computeConditionalQuantile(u, result));
-    result.stack(q);
-  }
+  if (dimension_ == 1) return computeQuantile(sequence.generate(size).asPoint());
+  // Use conditional CDF inversion in the nD case
+  Sample result(size, dimension_);
+  for (UnsignedInteger i = 0; i < size; ++ i)
+    result[i] = computeSequentialConditionalQuantile(sequence.generate());
```

Comment:
```
Point realization(computeSequentialConditionalQuantile(sequence.generate()));
for (UnsignedInteger j = 0; j < dimension_; ++ j)
   result(i, j) = realization[j]
```
We should avoid using `Sample[]` when possible

Reply:
Are you sure it will be more efficient? At least it will be less concise...

Reply:
I will do some tests

Reply:
@sofianehaddad It will be interesting. All these micro-benchmarks highlight Potential loss of performance. For this specific point, I finished to change all the call to Sample.operator[]() into calls to std::copy right now!

Reply:
I think it is not a priority. I pointed it to keep it in mind